src/cdomains/apron/sharedFunctions.apron.ml

```diff
@@ -285,11 +287,38 @@ struct
     !expr
 
 
+  let lcm_den linexpr1 =
+    let exception UnsupportedScalar 
+    in
+    let frac_of_scalar scalar =
+      if Scalar.is_infty scalar <> 0 then (* infinity means unbounded *)
+        None
+      else match scalar with
+        | Float f -> if Stdlib.Float.is_integer f then Some (Q.of_float f) else None
```

Comment:
According to [Zarith documentation](https://antoinemine.github.io/Zarith/doc/latest/Q.html#VALof_float), `Q.of_float` is exact, so shouldn't this also work for non-integers?